### PR TITLE
pre-install minimal requirment python packages on vscode

### PR DESCRIPTION
**Pull request by atheo89**
Related to: https://github.com/opendatahub-io/notebooks/issues/345 

## Description

This PR introduce a minimal list of python packages that are required for data sciences and programming workloads.

**NOTE:** Will follow up a separated issue/pr to incorporate the standard database clients https://github.com/opendatahub-io/notebooks/issues/372

## How Has This Been Tested?
1. Spin up the vscode image generated from this PR
2. Open a terminal inside the container and run `pip list` and ensure there is a long list with packages
3. Test some of the installed packages, by creating a .py or .ipynb file, add the following code:
```
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Create a sample DataFrame
data = {'A': np.random.rand(100),
        'B': np.random.randn(100),
        'C': np.linspace(0, 1, 100)}

df = pd.DataFrame(data)

# Perform some operations using NumPy
df['D'] = np.sin(df['A']) + np.cos(df['B'])

# Plotting with Matplotlib
plt.figure(figsize=(10, 6))

plt.subplot(2, 2, 1)
plt.scatter(df['A'], df['B'])
plt.title('Scatter Plot')

plt.subplot(2, 2, 2)
plt.plot(df['C'], df['D'], color='red')
plt.title('Line Plot')

plt.subplot(2, 2, 3)
plt.hist(df['B'], bins=20, color='green', alpha=0.7)
plt.title('Histogram')

plt.subplot(2, 2, 4)
plt.boxplot(df[['A', 'C']])
plt.title('Boxplot')

plt.tight_layout()
plt.show()
```
4. You shall see the following plot:
image

5. In case of issue check which interpreter you are using 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by rkpattnaik780 — APPROVED** [review]
LGTM! Able to run code snippets that use the added libraries.

/approve

**Comment by atheo89** [comment]
Glad to hear that! :slightly_smiling_face: 

> Also, we have a topic of where to put VSCode extensions and make them available without download ... but that is a different story. 

We are cooking something as well for this! Check out here: https://github.com/opendatahub-io/notebooks/pull/347

> Best regards from snowy CH.

Greetings from Turin on the other side of the Alps!

File: codeserver/ubi9-python-3.9/Pipfile [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+scipy = "~=1.11.4"
+skl2onnx = "~=1.15.0"
+codeflare-sdk = "~=0.12.1"
+kfp = "~=1.8.22"
+kfp-tekton = "~=1.8.1"
+ipykernel = "~=6.26.0"
+
+# PyTorch packages
+tensorboard = "~=2.15.1"
+torch = {version = "~=2.1.1", index = "pytorch"}
```
Comment (harshad16) [inline codeserver/ubi9-python-3.9/Pipfile:35]: any reason for only installing torch and not tensorflow.
on hindsight , i feel we should not install any of these big packages.
and only stick with basic packages like

boto3, matplotlib, pandas, numpy, scipy

**Review by harshad16 — CHANGES_REQUESTED** [review]
Thanks for the excellent work.
After closing checking, feels like we are install more packages then intended

File: codeserver/ubi9-python-3.9/Pipfile (hunk 1 above)
Comment (shalberd) [reply codeserver/ubi9-python-3.9/Pipfile:35]: yes, there is a similar issue in @guimou recent changes to the contrib datascience UI pathon packages, they have all sorts of heavy nvidia* libraries in it, regardless of whether the base docker image is for Tensorflow or not. https://github.com/opendatahub-io-contrib/workbench-images/issues/48. Agreed that in a first step here, minimal might be good. And you could get these changes here in this PR into the snippets for codeserver VSCode [at contrib](https://github.com/opendatahub-io-contrib/workbench-images/tree/main), so users can then always add more of their own packages, if they want to, in own builds.

**Comment by guimou** [comment]
Yeah, the issue comes from the codeflare sdk, which has dependencies on
pytorch, then nvidia...
For contrib I wanted to introduce codeflare in all the images, as well as
Elyra/kfp-tekton, but I'm not sure it's a good idea.
I talked to the Codeflare team, they were supposed to review their
dependencies, I did not check yet if they managed to have less heavy
dependencies.
So I guess it will come down to another discussion: which images should
include Codeflare sdk?

File: codeserver/ubi9-python-3.9/Pipfile (hunk 1 above)
Comment (shalberd) [reply codeserver/ubi9-python-3.9/Pipfile:35]: answerered in contrib https://github.com/opendatahub-io-contrib/workbench-images/issues/48#issuecomment-1842844803

**Comment by shalberd** [comment]
@guimou to keep that aspect separate, I answered in [odh-contrib](https://github.com/opendatahub-io-contrib/workbench-images/issues/48#issuecomment-1842844803)

File: codeserver/ubi9-python-3.9/Pipfile (hunk 1 above)
Comment (atheo89) [reply codeserver/ubi9-python-3.9/Pipfile:35]: Thank you for reviewing this! 😊 I was in a dilemma about whether to include the most advanced packages or not. I agree with you; I'll omit them for now and stick to the basic data science  packages.

**Review by harshad16 — APPROVED** [review]
Thanks for the work, lgtm.

/lgtm
/approve

Thanks all

**Pull request merged**